### DISCUSSION TITLE
Re mounting issue in next 13

### DISCUSSION THREAD

**Discussion by shubhrastogi007** [vercel/next.js#48437] [Help]
### Summary

import { useRouter, useSearchParams, usePathname } from 'next/navigation';

export function useQueryParams() {
    const pathname = usePathname();
    const router = useRouter();
    const searchParams = useSearchParams()!;
    
    const updateQueryParams = ({ query }: { query?: Record<string, string> }) => {
        const newSearchParams = new URLSearchParams(searchParams);
        Object.entries(query ?? {}).forEach(([key, value]) => {
            value !== '' ? newSearchParams.set(key, value) : newSearchParams.delete(key)
        });
        if(pathname) {
            const updatedUrl = `${pathname}?${decodeURIComponent(newSearchParams.toString())}`;
            router.replace(updatedUrl);
        }
    }
    
    return { update: updateQueryParams };
}

```js
While using useQueryParams hook's update function, due to router.replace whole app is re mounting, any solution to prevent re mounting.
```

**Comment by zek** [comment]
We also have a similar problem. Any ideas?
